soda_machine.rb

```diff
@@ -7,12 +7,17 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.find { |soda| soda.brand == soda_brand }
   end
 
   def sell(soda_brand)
+    sold_soda = find_soda(soda_brand)
```

Comment:
Excellent job using your previously written method and not recreating work you have already done!